assets/js/utils/render-frontend.js

```diff
@@ -2,6 +2,44 @@
  * External dependencies
  */
 import { render } from 'react-dom';
+import { getSetting } from '@woocommerce/settings';
+import { SCHEMA_STORE_KEY } from '@woocommerce/block-data';
+import { useEffect, useRef } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { forEach } from 'lodash';
+
+/**
+ * Hydrate Rest API data from settings to reduce the number of API requests needed.
+ *
+ * @param {mixed} children Child components.
+ */
+const HydrateRestApiData = ( { children } ) => {
+	const { receiveRoutes, startResolution, finishResolution } = useDispatch(
+		SCHEMA_STORE_KEY
+	);
+	const { isResolving, hasFinishedResolution } = useSelect( ( select ) => {
+		const store = select( SCHEMA_STORE_KEY );
+		return store;
+	} );
+	const RestApiRoutes = useRef( getSetting( 'restApiRoutes' ) );
+
+	useEffect( () => {
+		if ( RestApiRoutes.current ) {
+			forEach( RestApiRoutes.current, ( routes, namespace ) => {
+				if (
+					! isResolving( 'getRoutes', [ namespace ] ) &&
+					! hasFinishedResolution( 'getRoutes', [ namespace ] )
+				) {
+					startResolution( 'getRoutes', [ namespace ] );
+					receiveRoutes( routes, [ namespace ] );
+					finishResolution( 'getRoutes', [ namespace ] );
+				}
+			} );
+		}
+	}, [ RestApiRoutes ] );
```

Comment:
Refs don't get used as dependencies because they never change instance.  This means effectively this dependency here is the same as `[]`.